core/src/main/java/org/apache/carbondata/core/datamap/dev/AbstractDataMapWriter.java

```diff
@@ -0,0 +1,110 @@
+/*
```

Comment:
Can you explain , why change "DataMapWriter.java" to "AbstractDataMapWriter.java",  for easier supporting uses to customize other type of datamapwriter?

Reply:
Changed to Abstract class to enforce the user to pass the needed parameters through the constructor. And also the concrete method `commitFile` is added to this class.